### DISCUSSION TITLE
Issues trying to PXE boot using the FOG server.

### DISCUSSION THREAD

**Discussion by cecil29000** [PartialVolume/shredos.x86_64#208] [General]
We have a FOG server already in place that our clients boot to for imaging which loads up using iPXE over UEFI.  I have added an advanced menu for iPXE that looks for the ShredOS img file.  The file downloads correctly, but does not load ShredOS, giving an error of "Exec format error". Even if I extract all the files for ShredOS and point to the bootx64.efi, it will download and drop me to a grub menu where it does not appear to be pulling in the actual shredos file (even when downloading the file using initrd).  Is it possible to PXE boot ShredOS by chaining it from FOG?

**Comment by PartialVolume** [comment]
How are you storing the files. I've got a feeling the bootx64.efi expects the shredos image file to be located in /boot, if you are unpacking it into a sub folder off root that could be why it drops into grub. The ShredOS EFI contains all the necessary modules for PXE booting so should work as long as the files are off root which may not be ideal for PXE booting. If you find it does work like that then I may have to change the efi code so it uses relative rather than absolute directories.

**Comment by cecil29000** [comment]
I have tried many ways, including your suggestion of being in the root of the drive.  I tried setting it to use both TFTP and HTTP and the files download but still drops me into grub.  here is my current menu, I may be missing an argument somewhere though, I am not that well versed in iPXE and examples seem to be hard to come by.

I have tried just chaining the file in with this:
:ShredOS
chain http://${fog-ip}/EFI/BOOT/bootx64.efi

I have also tried downloading all the files, which is probably incorrect:
:ShredOS
kernel http://${fog-ip}/EFI/BOOT/bootx64.efi -c http://${fog-ip}/EFI/BOOT/grub.cfg
initrd http://${fog-ip}/boot/shredos
boot

When dropped into grub, doing an ls does not show any sort of drive to load shredos from.  I also cannot set grub to look to an http address with linux (http,10.1.0.27)/boot/shredos as it does not look like the http module is loaded in.

Is there some other way the iPXE menu should be structured to let this all line up?

I appreciate the help!

**Comment by PartialVolume** [comment]
I'll setup a FOG server and see if I can see what the issue is.

Does the client computer you are using boot a vanilla ShredOS from USB? Just want to check there aren't any issues with the client computer

**Comment by cecil29000** [comment]
Thank you for looking into this!  We are running FOG on Ubuntu 20.04 with the latest dev-branch version 1.5.10.15.

The computer does load up when using a USB key with the UEFI boot, so the files I have are definitely the correct ones

**Comment by PartialVolume** [comment]
As you mentioned the http module is not currently included in the ShredOS EFI, I can add that for the next release. However ShredOS will boot over pxe if you use tftp instead of http assuming you can just swap out http with tftp in FOG. Maybe a question for the FOG forum.

I know I said I would install and try this but I then looked at the documentation and realised it would take up too much time. Sorry.

**Comment by cecil29000** [comment]
Thank you for looking into this!  We will just keep using it on USB keys for now until we can revisit in the future.

**Comment by Chris87871** [comment]
:ShredOS
kernel http://${fog-ip}/usos/boot/bzImage -c http://${fog-ip}/usos/boot/grub/grub-eltorito.img
boot
I got shred os to boot in fog like this but if I hit enter the command line glitches and appears. Any ideas why?

**Reply by PartialVolume** [reply]
Have you tried using tftp:// instead of http:// ?

**Reply by Chris87871** [reply]
Yes same issues  booting with with tftp instead of http. Usb works fine.

**Comment by jamesaepp** [comment]
My comment here may be of interest: https://github.com/PartialVolume/shredos.x86_64/discussions/148#discussioncomment-10939106

**Reply by Chris87871** [reply]
Your setting worked. Thanks. All I had to modify was my boot path.
